harness/features/variable.local.test.ts

```diff
@@ -0,0 +1,325 @@
+        id: '638681f059f1b81cc9e6c7fa',
+        defaultValue: false,
+        variationOn: true,
+        variationOff: false
+    },
+    JSON: {
+        key: 'json-var',
+        id: '638681f059f1b81cc9e6c7fd',
+        defaultValue: {},
+        variationOn: {
+            'facts': true
+        },
+        variationOff: {
+            'facts': false
+        }
+    }
+}
+
+describe('Variable Tests - Local', () => {
+    forEachSDK((name) => {
+        let url: string
+        const capabilities: string[] = SDKCapabilities[name]
+        const clientId: string = uuidv4()
+        const mockServerUrl 
+            = `http://${process.env.DOCKER_HOST_IP ?? 'host.docker.internal'}:${global.__MOCK_SERVER_PORT__}`
+
+        const baseURL = `${mockServerUrl}/client/${clientId}`
+        const variationOnUser = { location: '', user_id: 'user1' }
+        const noVariationUser = { location: '', user_id: 'user3' }
+        const invalidUser = { location: '', user_id: '' }
+        
+        beforeAll(async () => {
+            url = getConnectionStringForProxy(name)  
+
+            // create client          
+            await createClient(url, clientId, 'dvc_server_test_token', {
+                baseURLOverride: baseURL,
+                eventFlushIntervalMS: 500
+            })
+
+            // create users
+            variationOnUser.location = (
+                await createUser(url, { user_id: variationOnUser.user_id, customData: { 'should-bucket': true } })
+            ).headers.get('location')
+
+            noVariationUser.location = (
+                await createUser(url, { user_id: noVariationUser.user_id })
+            ).headers.get('location')
+
+            invalidUser.location = (
+                await createUser(url, { name: 'invalid' })
+            ).headers.get('location')
+        })
+
+        describeIf(capabilities.includes(Capabilities.local))(name, () => {
+            describe('uninitialized client', () => {
+                it('should return default value if client is uninitialized',  async () => {
+                    const variableResponse = await callVariable(
+                        clientId, url, variationOnUser.location, 'string-var', 'string default'
+                    )
+                    const variable = await variableResponse.json()
+                    expectDefaultValue(variable, 'string default')
+                })
+
+                it('should return default value if user is invalid',  async () => {    
+                    const variableResponse =
+                        await callVariable( clientId, url, invalidUser.location, 'string-var', 'string default')
+                    const variable = await variableResponse.json()
+                    expectDefaultValue(variable, 'string default')
+                })
+            })
+
+            describe('initialized client', () => {
+                beforeAll(async () => {
+                    // mock endpoints
+                    scope
```

Comment:
seems kind of weird to provide this request mock after the client has already been created earlier, what if it tries to hit this endpoint before this is set up?

Reply:
true, i moved it to the beforeall